### ISSUE TITLE
Configuring SES to an existing userpool removes already added lambdatriggers

### ISSUE THREAD

**Issue by severi** [serverless/serverless#7510]
# Bug Report

## Description

1. What did you do?
Added `EmailConfiguration` to my existing user pools Cloudformation to send cognito emails via SES (https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-email.html). Serverless functions were configured using the `existing: true` parameter.

1. What happened?
The existing lambda triggers I had configured via the Serverless Framework disappeared from the userpool. If I create a new stack from scratch it seems to create the lambdas correctly. I suppose this is somehow related to me using the `existing: true` flag

I tried to circumvent the problem by renaming the lambda trigger function names, but this also failed

```
ServerlessError: An error occurred: CustomMessageTriggerCustomCognitoUserPool1 - Failed to delete resource. User: arn:xxxxxxx:assumed-role/yyyy-IamRoleCustomResourcesLa-zzzzz/stackname-custom-resource-existing-cup is not authorized to perform: lambda:RemovePermission on resource: arn:aws:lambda:xxxxx:12314123:function:stackname-customMessageTrigger See details in CloudWatch Log:.
```

1. What should've happened?
The lambda triggers should not have disappeared.

1. What's the content of your `serverless.yml` file?
```
service: example

package:
  individually: true

plugins:
  - serverless-bundle

custom:
  stage: ${opt:stage, self:provider.stage}

provider:
  name: aws
  runtime: nodejs12.x
  stage: dev
  region: eu-west-1
  environment:
    stage: ${self:custom.stage}

functions:
  preSignUp:
    handler: services/cognito/preSignUp.handler
    events:
      - cognitoUserPool:
          pool: ${self:custom.stage}-user-pool
          trigger: PreSignUp
          existing: true
  customMessage:
    handler: services/cognito/customMessage.handler
    events:
      - cognitoUserPool:
          pool: ${self:custom.stage}-user-pool
          trigger: CustomMessage
          existing: true

resources:
  - ${file(resources/cognito-user-pool.yml)} # creates the userpool
  - ${file(resources/cognito-identity-pool.yml)}
```

**Comment by severi** [comment]
Seems like one way to circumvent this problem is to remove the lambda triggers from the deployment that adds the SES configs to the userpool via cloudformation, and then re-enable the triggers and rerun the deployment

**Comment by medikoo** [comment]
@severi are you stating, that reconfiguring triggers to _existing_ pools. Will remove existing triggers (together with pool that was created), and not set new triggers for configured _existing_ pools?

> I tried to circumvent the problem by renaming the lambda trigger function names, but this also failed

This error looks as permissions error signaling that entity with which service is deployed has not necessary rights to follow up with changes.

**Comment by severi** [comment]
> @severi are you stating, that reconfiguring triggers to _existing_ pools. Will remove existing triggers (together with pool that was created), and not set new triggers for configured _existing_ pools?
> 
> > I tried to circumvent the problem by renaming the lambda trigger function names, but this also failed
> 
> This error looks as permissions error signaling that entity with which service is deployed has not necessary rights to follow up with changes.

I'm stating, that in a scenario in which you 
1) use an existing userpool (`existing: true`) that does not yet have SES enabled (EmailConfiguration). The user pool is created with Cloudformation and deployed via serverless framework.
2) and have configured lambda triggers for that user pool using the serverless famework
3) already have deployed all of this to AWS using the serverless framework

the lambda triggers will disappear, if you now now add the `EmailConfiguration`to your userpool and redeploy the stack with serverless framework. It will also casue the userpool to go into some sort of a limbo-state that (at least I) could not find a way to recover from. 

I was able to circumvent this problem by first commenting out the lambda triggers from the serverless.yml and deploying the stack (i.e. the triggers were removed), then added the EmailConfiguration-parameter to my userpools cloudformation template and deployed it (with `sls deploy`) and then uncommented the lambda triggers redeployed the stack (`sls deploy`) 

So to summarize: configuring/reconfiguring existing lambda triggers does not cause any issues, but enabling SES for the user pool that already has lambda triggers does

**Comment by medikoo** [comment]
So in short we may probably summarize, that adding `EmailConfiguration` to Cognito Pool that's referenced (via `existing: true`) in some service, will remove the triggers.

I wonder weather fact that you deploy this _existing_ cognito pools with same stack (via `resources` section) has some influence (will it also happen if cognito pool will be created externally outside of a stack?)

This looks as some AWS culprit, I wonder whether there's anything on Framework level we can improve here (?) (note that via _existing: true_ you mark that you handle creation and managment of given Cognito pool in your capacity)

**Comment by severi** [comment]
> So in short we may probably summarize, that adding `EmailConfiguration` to Cognito Pool that's referenced (via `existing: true`) in some service, will remove the triggers.
> 
> I wonder weather fact that you deploy this _existing_ cognito pools with same stack (via `resources` section) has some influence (will it also happen if cognito pool will be created externally outside of a stack?)
> 
> This looks as some AWS culprit, I wonder whether there's anything on Framework level we can improve here (?) (note that via _existing: true_ you mark that you handle creation and managment of given Cognito pool in your capacity)

True, it could be possible that this issue is caused by AWS.
However, my initial hunch was that the issue might be somehow related to the following statement about using existing userpools (copied from the serverless framework docs): 

> NOTE: Using the existing config will add an additional Lambda function and IAM Role to your stack. The Lambda function backs-up the Custom Cognito User Pool Resource which is used to support existing user pools.

I don't know the technical details why the framework is doing this or what is happening under the hood, but could this somehow be one part of the problem causing the issue in this case? As stated before, it seems like the userpool goes to some kind of a limbo state when it removes the lambdas, as they cannot be redeployed/fixed afterwards

**Comment by medikoo** [comment]
> I don't know the technical details why the framework is doing this or what is happening under the hood

Custom resources are used to fill CloudFormation limitations. If we cannot achieve needed resources setup purely via CF means, then we configure a custom resource which via direct AWS SDK calls ensures wanted setup.

It's hard for me to say wether it's an issue in custom resource configuration or on AWS side. Anyway help on that matter is welcome!

**Comment by mattnedrich** [comment]
We're experiencing the same issue. The thread isn't clear if this was able to be root caused as a serverless framework issue or cloudformation? Can anyone clarify?

**Comment by rally25rs** [comment]
Simply changing the `InviteMessageTemplate.EmailMessage` also causes this issue to occur. So if you ever change the Cognito email messages you end up having to fix your lambda triggers after deploy.

**Comment by pgrzesik** [comment]
Hello @rally25rs - I've managed to reproduce this issue as a part of https://github.com/serverless/serverless/issues/9635 I currently don't have the capacity to introduce the fix, but PRs are welcome. I plan to work on it at some point in the future but I don't have ETA

**Comment by medikoo** [comment]
I'm going to close it as duplicate of #9635 please follow it for the progress. Also PR that addresses that is welcome (see https://github.com/serverless/serverless/issues/9635#issuecomment-873998830 on proposed solution)
